### PR TITLE
Add sanitizers to DynamicPoolMap and DynamicPoolList

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

File: src/umpire/util/memory_sanitizers.hpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#ifndef UMPIRE_memory_sanitizers_HPP
+#define UMPIRE_memory_sanitizers_HPP
+
+#if defined(__clang__)
```
Comment (mcfadden8) [inline src/umpire/util/memory_sanitizers.hpp]: Don't we want to make this optionally be compiled in?  I thought I saw something about a 73% slowdown?

Comment (davidbeckingsale) [reply src/umpire/util/memory_sanitizers.hpp]: The ASAN macros themselves are no-ops when compiled without `-fsanitize=address`

**Comment by mcfadden8** [comment]
This looks great!  Would you add tests for this?  I'd like to see what it does for both stores and loads if possible.

**Comment by davidbeckingsale** [comment]
How do you think it would best be tested?

File: src/umpire/util/memory_sanitizers.hpp (hunk 1 above)
Comment (davidbeckingsale) [reply src/umpire/util/memory_sanitizers.hpp]: I will add an option to guard it also, since it adds a conditional for the `Platform` check.

**Comment by mcfadden8** [comment]
With a signal handler?

**Pull request merged**